src/Albany_StateManager.cpp

```diff
@@ -575,14 +575,9 @@ Albany::StateManager::registerSideSetStateVariable_collapsed(
     else if (dl->rank() == 4)  // node tensor
       nodalDataBase->registerVectorState(
           stateName, stateRef.dim[1] * stateRef.dim[2]);
-  } else {
-
-    TEUCHOS_TEST_FOR_EXCEPTION(
-        true,
-        std::logic_error,
-        "Error! The given layout does not appear to be that of a collapsed side set "
-        "field.\n");
-
+  }
+  else {
```

Comment:
Why not erroring out? It seemed appropriate, if the input was unexpected.

Reply:
The exception that I removed here was copy and pasted from the non-collapsed implementation and only throws if there is a layout with only one field. Since the old sideset layouts have to have two fields (Cell, Side) at minimum, there was no way it could have only one field. However, with the collapsed layout, it is fine to have a single field (Side). From what I could gather, for a single field layout, all we need to do here is get the dimensions.

Reply:
I'm just weary of 'else' statements, cause they catch whatever didn't fit in the previous if(s). Maybe we can put a test for exception, ensuring that the rank is indeed 1? Right now, you didn't add, say, `qp_tensor3` as a collapsed layout to Albany::Layout. But if tomorrow someone did it, and used it, adding a check in here might save hours of debugging.

Reply:
Note: alternatively, you could add a check on the layout rank at the beginning of the fcn, and support only rank<=4 for now.

Reply:
That's a good point. I'll rewrite this to make sure nothing gets through the cracks. I assume that means an else-if for rank 1 and then throw an exception in the else but I'll double check to make sure that covers all cases.

Reply:
I revisited this section of the code to see what needed to be done here. The else here is covering the case where stateRef.entity is not nodal. There isn't really anything in this if-else that depends on layout size so I'm not sure it actually makes sense to do an exception here based on layout rank beyond checking for the bad case rank == 1 which is no longer relevant for collapsed layouts. 

It seems like to me that there are already exceptions earlier in the function to make sure we aren't getting unexpected input so as far as I can tell nothing needs to be added here. Let me know if I am missing something here since I am not super familiar with the stateRef entity stuff.